aptos-move/block-executor/src/errors.rs

```diff
@@ -17,14 +18,40 @@ pub enum IntentionalFallbackToSequential {
     /// We defensively check resource group serialization error in the commit phase.
     /// TODO: should trigger invariant violation in the transaction itself.
     ResourceGroupSerializationError(String),
+    /// Execution of a thread yields a non-recoverable error from the VM. It it happens
+    /// during parallel execution, an error will be logged and sequential fallback taken.
+    FatalVMError(String),
```

Comment:
Do we need these variants? We can have just an `IntentionalFallbackToSequential` and the caller will log whether it is a module read-write error, resource group, etc.? Or am I missing something?

Reply:
Fair enough from returning from parallel point of view, but we still have to propagate the errors to the main worker loop?